src/components/mutations/useUpdateHypothesisMutation.js

```diff
@@ -1,61 +1,19 @@
 import { useMutation, useQueryClient } from "@tanstack/react-query";
 import { updateHypothesis } from "../../api/services/hypothesis";
 import { createIncidentQueryKey } from "../../api/query-hooks";
-
-const updateHypothesisInIncident = (incident, id, values) =>
-  incident.map((i) => ({
-    ...i,
-    hypothesis: i.hypothesis.map((h) => {
-      if (h.id === id) {
-        return {
-          ...h,
-          ...values
-        };
-      }
-      return h;
-    })
-  }));
+import { toastError } from "../Toast/toast";
 
 export const useUpdateHypothesisMutation = (options = {}) => {
-  const { incidentId = "incorrect_incident_id" } = options;
+  const { incidentId } = options;
   const queryClient = useQueryClient();
   const incidentQueryKey = createIncidentQueryKey(incidentId);
 
   return useMutation(({ id, params }) => updateHypothesis(id, params), {
-    onMutate: async (data) => {
-      const previousIncident = queryClient.getQueryData(incidentQueryKey);
-
-      if (!previousIncident || !previousIncident[0]) {
-        return { previousIncident };
-      }
-
-      const newIncident = updateHypothesisInIncident(
-        previousIncident,
-        data.id,
-        data.params
-      );
-
-      queryClient.setQueryData(incidentQueryKey, newIncident);
-
-      return { previousIncident, newIncident };
-    },
-    onSuccess: (resp, variables) => {
-      const previousIncident = queryClient.getQueryData(incidentQueryKey);
-
-      if (previousIncident && previousIncident[0]) {
-        queryClient.setQueryData(
-          incidentQueryKey,
-          updateHypothesisInIncident(
-            previousIncident,
-            variables.id,
-            variables.params
-          )
-        );
-      }
+    onSuccess: () => {
+      queryClient.refetchQueries(incidentQueryKey);
     },
-    onError: (err, newData, context) => {
-      queryClient.setQueryData(incidentQueryKey, context.previousIncident);
-      queryClient.invalidateQueries(createIncidentQueryKey(incidentId));
+    onError: (err) => {
```

Comment:
Doesn't Axios already handle these APIs errors or only for GET?

Reply:
Yeah, we already have a global error handler added in one of the response interceptor. This is unnecessary here, I'll remove this.